systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -1450,6 +1457,157 @@ void testClientsCACertRenew() {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    void testCustomClusterCAClientsCA() {
+        generateAndDeployCustomStrimziCA();
+        LOGGER.info("Check ClusterCA and ClientsCA certificates.");
+        X509Certificate clientsCert = SecretUtils.getCertificateFromSecret(kubeClient().getSecret(KafkaResources.clientsCaCertificateSecretName(clusterName)), "ca.crt");
+        X509Certificate clusterCert = SecretUtils.getCertificateFromSecret(kubeClient().getSecret(KafkaResources.clusterCaCertificateSecretName(clusterName)), "ca.crt");
+        assertThat("Generated ClientsCA does not have expected Issuer: " + clientsCert.getIssuerDN(),
+                SystemTestCertManager.containsAllDN(clientsCert.getIssuerX500Principal().getName(), STRIMZI_INTERMEDIATE_CA));
+        assertThat("Generated ClientsCA does not have expected Subject: " + clientsCert.getSubjectDN(),
+                SystemTestCertManager.containsAllDN(clientsCert.getSubjectX500Principal().getName(), STRIMZI_TEST_CLIENTS_CA));
+
+        assertThat("Generated ClusterCA does not have expected Issuer: " + clusterCert.getIssuerDN(),
+                SystemTestCertManager.containsAllDN(clusterCert.getIssuerX500Principal().getName(), STRIMZI_INTERMEDIATE_CA));
+        assertThat("Generated ClusterCA does not have expected Subject: " + clusterCert.getSubjectDN(),
+                SystemTestCertManager.containsAllDN(clusterCert.getSubjectX500Principal().getName(), STRIMZI_TEST_CLUSTER_CA));
+
+        LOGGER.info(" Deploy kafka with new certs/secrets.");
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(clusterName, 3, 3)
+            .editSpec()
+                .withNewClusterCa()
+                    .withGenerateCertificateAuthority(false)
+                .endClusterCa()
+                .withNewClientsCa()
+                    .withGenerateCertificateAuthority(false)
+                .endClientsCa()
+                .editKafka()
+                    .withNewListeners()
+                        .addNewGenericKafkaListener()
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                            .withPort(9092)
+                            .withTls(false)
+                        .endGenericKafkaListener()
+                        .addNewGenericKafkaListener()
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                            .withPort(9093)
+                            .withTls(true)
+                            .withNewKafkaListenerAuthenticationTlsAuth()
+                            .endKafkaListenerAuthenticationTlsAuth()
+                        .endGenericKafkaListener()
+                    .endListeners()
+                .endKafka()
+            .endSpec()
+            .build());
```

Comment:
Maybe I'm completely wrong (if yes, don't mind me), but when you firstly create the certs (firstly you delete the "current one", if there are some) and then create Kafka cluster, I guess the certs will be overwritten (when the name is same like from CO).

I would do this:
1) create Kafka cluster
2) delete the certs and create new (or make certs with different names as we discussed offline)
3) then try to send/receive messages and do the asserts

Reply:
There are not created anew/overwritten - because 
1) Tests would already be failing on asserts, because of issuer/subject DNs would be different as expected
2) ```.withNewClusterCa()
                    .withGenerateCertificateAuthority(false)
                .endClusterCa()
                .withNewClientsCa()
                    .withGenerateCertificateAuthority(false)
                .endClientsCa()```
This should prohibit the creation of "default" strimzi certs - https://strimzi.io/docs/operators/master/using.html#installing-your-own-ca-certificates-str (at the very end of this part of docs, right above 9.2)

Reply:
Yeah, that's true, didn't saw that ` .withGenerateCertificateAuthority(false)` but still - I would use different cert's names. It's up to you.

Reply:
I'm not sure if I undertsand it correctly what are you asking for @im-konge ... but the CA secrets need to follow the operator naming. You cannot give them custom names.

Reply:
Ah okay, I thought that we can give them different name and than use it. Thanks for correction :slightly_smiling_face: